api/module.go

```diff
@@ -27,6 +28,18 @@ func (m ModuleController) DeployModule(c *gin.Context) {
 		return
 	}
 
+	if !data.Completed {
```

Comment:
We could probably break this block out into a helper function that can then be called once from each controller rather than duplicating the code twice. Granted, that requires updating the Helpers interface with a new function and regenerating the mocks with `mockery`.

(NOTE TO SELF: Add mockery instructions to README)

Reply:
New file in `lib/helpers` called `pipeline-status.go` with content:

```go
package helpers

import (
    "errors"

    "github.com/voxpupuli/webhook-go/lib/parsers"
)

func (h *Helper) CheckPipelineStatus(data *parsers.Data, deployOnErr bool) error {
        if !data.Completed {
		return errors.New("Received webhook but job is not complete. Ignoring.")
	} else if !data.Succeed && !pipelineConf.DeployOnError {
		return errors.New("Received webhook but job failed. Ignoring.")
	}
	return nil
}
```

module.go/controller.go:

```go
if err := helpers.CheckPipelineStatus(*data, pipelineconf.DeployOnError); err != nil {
                c.JSON(http.StatusAccepted, gin.H{"message": fmt.Sprintf("%v", err)})
		log.Debug(fmt.Sprintf("%v", err))
		c.Abort()
		return
}
```

Reply:
My code was written before the helpers refactoring. i'll update it, thanks! :smile: